app/src/main/java/com/android/unio/model/events/EventListViewModel.kt

```diff
@@ -0,0 +1,60 @@
+package com.android.unio.model.events
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+/**
+ * ViewModel class that manages the event list data and provides it to the UI. It uses an
+ * [EventRepository] to load the list of events and exposes them through a [StateFlow] to be
+ * observed by the UI.
+ *
+ * @property repository The [EventRepository] that provides the events.
+ */
+class EventListViewModel(private val repository: EventRepository) : ViewModel() {
+
+  /**
+   * A private mutable state flow that holds the list of events. It is internal to the ViewModel and
+   * cannot be modified from the outside.
+   */
+  private val _events = MutableStateFlow<List<Event>>(emptyList())
+
+  /**
+   * A public immutable [StateFlow] that exposes the list of events to the UI. This flow can only be
+   * observed and not modified.
+   */
+  val events: StateFlow<List<Event>> = _events
+
+  /** Initializes the ViewModel by loading the events from the repository. */
+  init {
+    loadEvents()
+  }
+
+  /** Loads the list of events from the repository and updates the internal [MutableStateFlow]. */
+  fun loadEvents() {
+    _events.value = repository.getEvents()
```

Comment:
When at a later time the `EventRepositoryFirestore` is integrated, this will not work, as `repository.getEvents` is asynchronous. Consider changing this to use a callback function.

Reply:
Yes thank you ! Copilot proposed it to me, but as I didn't know how EventRepositoryFirestore would be made, i didn't do it